compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -258,7 +258,18 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 Ok(ty::Visibility::Restricted(DefId::local(CRATE_DEF_INDEX)))
             }
             ast::VisibilityKind::Inherited => {
-                Ok(ty::Visibility::Restricted(parent_scope.module.normal_ancestor_id))
+                if self.parent_scope.module.kind.is_enum() {
+                    // Anything in an enum has the visibility of the enum.
+                    Ok(*self
+                        .r
+                        .visibilities
+                        .get(&self.parent_scope.module.normal_ancestor_id.expect_local())
+                        .unwrap())
```

Comment:
nit: rustfmt really mangles this
```suggestion
                    // Anything in an enum has the visibility of the enum.
                    let parent_enum = self.parent_scope.module.normal_ancestor_id.expect_local();
                    Ok(*self.r.visibilities.get(&parent_enum).unwrap())
```

Reply:
Fixed in 26886f8.